modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1955,21 +2031,24 @@ def _materialize(self):
 
         Returns
         -------
-        pyarrow.Table
+        DbTable or pyarrow.Table
         """
-        assert (
-            self._force_execution_mode != "lazy"
-        ), "Unexpected execution triggered on lazy frame!"
+        mode = self._force_execution_mode
+        assert mode != "lazy", "Unexpected execution triggered on lazy frame!"
+
+        if isinstance(self._op, FrameNode):
+            return self._op.execute_arrow()
 
-        if self._force_execution_mode != "hdk" and self._can_execute_arrow():
+        if (
+            mode == "arrow"
+            or not self._op.can_execute_hdk()
+            or (self._can_execute_arrow() and mode != "hdk")
```

Comment:
I see that _force_execution_mode description only mentions "lazy" and "arrow", but there is also "hdk", right?

Reply:
Added "hdk" to the description.